OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -181,7 +193,6 @@ void Attack(Actor self, Actor targetActor, bool allowMove)
 			var target = Target.FromActor(targetActor);
 			self.SetTargetLine(target, Color.Red, false);
 
-			var activeAttackBases = attackBases.Where(Exts.IsTraitEnabled);
```

Comment:
I'm caching the enumerable itself, not the results. It will get re-evaluated each time.